internal/handlers/sqlite/msg_renamecollection.go

```diff
@@ -16,13 +16,143 @@ package sqlite
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonparams"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgRenameCollection implements HandlerInterface.
 func (h *Handler) MsgRenameCollection(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/2760
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	var err error
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	// implement dropTarget param
+	// TODO https://github.com/FerretDB/FerretDB/issues/2565
+	if err = common.UnimplementedNonDefault(document, "dropTarget", func(v any) bool {
+		b, ok := v.(bool)
+		return ok && !b
+	}); err != nil {
+		return nil, err
+	}
+
+	ignoredFields := []string{
+		"writeConcern",
+		"comment",
+	}
+	common.Ignored(document, h.L, ignoredFields...)
+
+	command := document.Command()
+
+	oldName, err := common.GetRequiredParam[string](document, command)
+	if err != nil {
+		from, _ := document.Get(command)
+
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrBadValue,
+			fmt.Sprintf("collection name has invalid type %s", commonparams.AliasFromType(from)),
+			command,
+		)
+	}
+
+	newName, err := common.GetRequiredParam[string](document, "to")
+	if err != nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			"'to' must be of type String",
+			command,
+		)
+	}
+
+	oldDBName, oldCName, err := commonparams.SplitNamespace(oldName, command)
+	if err != nil {
+		return nil, err
+	}
+
+	newDBName, newCName, err := commonparams.SplitNamespace(newName, command)
+	if err != nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrInvalidNamespace,
+			fmt.Sprintf("Invalid target namespace: %s", newName),
+			command,
+		)
+	}
+
+	// support cross-database rename
+	// TODO https://github.com/FerretDB/FerretDB/issues/2563
+	if oldDBName != newDBName {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrNotImplemented,
+			"Command renameCollection does not support cross-database rename",
+			command,
+		)
+	}
+
+	if oldCName == newCName {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrIllegalOperation,
+			"Can't rename a collection to itself",
+			command,
+		)
+	}
+
+	db, err := h.b.Database(oldDBName)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseNameIsInvalid) {
+			msg := fmt.Sprintf("Invalid namespace specified '%s'", oldName)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, command)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+	defer db.Close()
```

Comment:
This needs to be removed on https://github.com/FerretDB/FerretDB/pull/3327 is merged.

Reply:
It was merged

Reply:
Perfect!